### DISCUSSION TITLE
Search notes in folders by title

### DISCUSSION THREAD

**Discussion by cdaein** [xwmx/nb#327] [Q&A]
Is there a way to search every note in a notebook, including any sub-folders in the same notebook by title?

For example, if my notes are:
```sh
[folder/3] Cherry
[2] Banana
[1] Apple
```
`nb ls cherry` will not return `[folder/3]`.

```sh
! Not found: cherry
```

`nb q cherry -l` will include [folder/3] but will also include other notes if they contain the text "cherry" inside.

```sh
[1] Apple
[2] Banana
[folder/3] Cherry
```

I was doing `nb q cherry -l | grep -i cherry` but was hoping to find a better solution.

**Comment by xwmx** [comment]
There is currently an undocumented `--recursive` / `--tree` option that will return `[folder/3]` when running `nb ls cherry`. This option is still undocumented because some advanced `list` features don't work fully with it yet, but it should work fine for `nb ls`.

**Reply by cdaein** [reply]
Thank you for the response.
For some reason, when I `nb ls cherry --tree`, it not only prints `[folder/3]` but also prints other entries that do not contain `cherry` as text. It also prints the same list twice.
